File: src/gestorAplicacion/logistica/Asiento.java
package gestorAplicacion.logistica;

import java.io.*;

public class Asiento implements Serializable{
    
    private static final long serialVersionUID= 1L;
    private int numeroAsiento;
    private Boolean disponibilidad;
    private Avion avion;
    private Clase clase;


    //constructores
    public Asiento(int numero_asiento, Avion avion, Clase clase){
        this.numeroAsiento = numero_asiento;
        this.disponibilidad = true;
        this.avion = avion;
        this.clase = clase;
    }

    
    //methods
    @Override
	public String toString() {
        if(this.clase == Clase.PREMIUM){
            String txt = this.numeroAsiento+"PR";
            return txt;
        }else if(this.clase == Clase.EJECUTIVA){
            String txt = this.numeroAsiento +"EJ" ;
            return txt; 
        }else{
            String txt = this.numeroAsiento+"EC";
            return txt;
        }
	}
    

    //getter and setter
    public int getNumeroAsiento() {
        return numeroAsiento;
    }
    public void setNumeroAsiento(int numero_asiento) {
        this.numeroAsiento = numero_asiento;
    }
    public Boolean getDisponibilidad() {
        return disponibilidad;
    }
    public void setDisponibilidad(Boolean disponibilidad) {
        this.disponibilidad = disponibilidad;
    }
    public Avion getAvion() {
        return avion;
    }
    public void setAvion(Avion avion) {
        this.avion = avion;
    }
    public Clase getClase() {
        return clase;
    }
    public void setClase(Clase clase) {
        this.clase = clase;
    }
}

File: Python/src/gestorAplicacion/logistica/Asiento.py
class Asiento:

    # constructores
    def __init__(self, numero_asiento, avion, clase):
        self.numeroAsiento = numero_asiento
        self.disponibilidad = True
        self.avion = avion
        self.clase = clase

    # methods
    def __str__(self):
        if self.clase == Clase.PREMIUM:
            txt = self.numeroAsiento+"PR"
            return txt
        elif self.clase == Clase.EJECUTIVA:
            txt = self.numeroAsiento+"EJ"
            return txt
        else:
            txt = self.numeroAsiento+"EC"
            return txt

    # getter and setter
    def getNumeroAsiento(self):
        return self.numeroAsiento

    def setNumeroAsiento(self, numero_asiento):
        self.numeroAsiento = numero_asiento

    def getDisponibilidad(self):
        return self.disponibilidad

    def setDisponibilidad(self, disponibilidad):
        self.disponibilidad = disponibilidad

    def getAvion(self):
        return self.avion

    def setAvion(self, avion):
        self.avion = avion

    def getClase(self):
        return self.clase

    def setClase(self, clase):
        self.clase = clase